ksql-serde/src/main/java/io/confluent/ksql/serde/connect/DataTranslator.java

```diff
@@ -0,0 +1,27 @@
+/**
+ * Copyright 2018 Confluent Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ **/
+
+package io.confluent.ksql.serde.connect;
+
+import io.confluent.ksql.GenericRow;
+import org.apache.kafka.connect.data.Schema;
+import org.apache.kafka.connect.data.Struct;
+
+public interface DataTranslator {
+  GenericRow toKsqlRow(final Schema connectSchema, final Object connectData);
```

Comment:
just for my own education, what is the `connectData`? Why is it `Object`? I thought it would be a `Struct`??

Reply:
Its returned by the Converter API inside a SchemaAndValue. It _should_ always be a struct here. If the param was a Struct we'd have to first check the type from the caller. Its simpler/cleaner to just have the translator validate the types.